odd-platform-api/src/main/java/com/provectus/oddplatform/config/OAuthSecurityConfiguration.java

```diff
@@ -1,40 +1,75 @@
 package com.provectus.oddplatform.config;
 
+import com.provectus.oddplatform.auth.CognitoOidcLogoutSuccessHandler;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.core.Ordered;
 import org.springframework.core.annotation.Order;
+import org.springframework.core.env.Environment;
 import org.springframework.security.config.Customizer;
 import org.springframework.security.config.web.server.ServerHttpSecurity;
+import org.springframework.security.oauth2.client.oidc.web.server.logout.OidcClientInitiatedServerLogoutSuccessHandler;
+import org.springframework.security.oauth2.client.registration.ReactiveClientRegistrationRepository;
 import org.springframework.security.web.server.SecurityWebFilterChain;
+import org.springframework.security.web.server.authentication.logout.ServerLogoutSuccessHandler;
 import org.springframework.security.web.server.util.matcher.PathPatternParserServerWebExchangeMatcher;
 
 @Configuration
 @ConditionalOnProperty(value = "auth.type", havingValue = "OAUTH2")
 public class OAuthSecurityConfiguration {
+
+    private final String clientId;
+    private final String logoutUrl;
+    private final Boolean cognitoEnabled;
+
+    public OAuthSecurityConfiguration(Environment env) {
```

Comment:
Instead of injecting the Environment variable, can we use @ConfigurationProperties here?

Reply:
Sure